src/libopensc/pkcs15.c

```diff
@@ -2426,6 +2426,9 @@ sc_pkcs15_format_id(const char *str, struct sc_pkcs15_id *id)
 {
 	size_t len = sizeof(id->value);
 
+	if (id)
+		memset(id, 0, sizeof(struct sc_pkcs15_id));
```

Comment:
This breaks CAC completely. Removing these two lines restores CAC to working properly, and keeps NEO working too.